knowledge_repo/app/app.py

```diff
@@ -96,33 +97,12 @@ def show_traceback(self):
                 return render_template('error.html')
 
         @self.before_first_request
-        def update_index_and_typeahead_data():
-            """ Function to call all of the steps we want to run recurrently
-                Ideally, these functions would be in a background job, but
-                running them on each user's first request is a workable
-                solution for now.
-
-                Putting these functions in a function wrapper also helps
-                because before_first_request decorated functions cannot be
-                imported, so this style allows the functions to be used
-                across the app in addition to being run at first.
-            """
-
-            if not current_app.config.get('REPOSITORY_INDEXING_ENABLED', True):
-                return
-
-            update_index()
-
+        def ensure_excluded_tags_exist():
             # For every tag in the excluded tags, create the tag object if it doesn't exist
             excluded_tags = current_app.config.get('EXCLUDED_TAGS', [])
             for tag in excluded_tags:
-                tag_exists = (db_session.query(Tag)
-                                        .filter(Tag.name == tag)
-                                        .all())
-                if not tag_exists:
-                    tag_exists = Tag(name=tag)
-                    db_session.add(tag_exists)
-                    db_session.commit()
+                Tag(name=tag)
```

Comment:
The way I crafted the model for `Tag` objects means that tags are deduplicated as a matter of course. There can never be duplicate tags when done this way.